### DISCUSSION TITLE
Pyarmor 8 supports FreeBSD

### DISCUSSION THREAD

**Discussion by Soujanya459** [dashingsoft/pyarmor#1256] [Q&A] [answered]
pip install pyarmor==8.1.8
Collecting pyarmor==8.1.8
  Downloading pyarmor-8.1.8-py2.py3-none-any.whl (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 478.5 kB/s eta 0:00:00
ERROR: Could not find a version that satisfies the requirement pyarmor.cli.core~=2.1.8 (from pyarmor) (from versions: none)
ERROR: No matching distribution found for pyarmor.cli.core~=2.1.8

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: pip install --upgrade pip

```
python : 3.11.2
OS: FreeBSD amd64
pip:  22.3.1
```

Can you please let me know what is the issue here? is pyarmor not supported in FreeBSD-AMD64 OS ?

**Comment by jondy** [comment, marked as answer]
FreeBSD is not still not supported by Pyarmor8, please check the latest documentation.

**Comment by joonis** [comment]
Any plans when it will be supported?

**Comment by jondy** [comment]
Maybe in the latter half of the year, to support new platforms, for example, Android, arch linux(musl-c), and FreeBSD etc.

**Comment by Soujanya459** [comment]
can you give us atleast source code of pyarmor.cli.core~=2.1.8 which works in FreeBSD.. 

please suggest quick workaround... as we are dependent on this pyarmor 8.1.8 version..

**Comment by jondy** [comment]
@Soujanya459 If `pyarmor.cli.core~=2.1.8` work, why not upgrade Pyarmor to 8.2.0? It should work too.

**Comment by Soujanya459** [comment]
No No .. I said Pyarmor.cli.core~=2.1.8 doesn't work in FreeBSD that's why Pyarmor8 is failing .. 

So can you please give us workaround/source code to build a binary of pyarmor, and make it work in FreeBSD platform.

Please give us fix for the issue as we need pyarmor8 in FreeBSD for our product..

```
# pip install pyarmor==8.2.0
Collecting pyarmor==8.2.0
  Downloading pyarmor-8.2.0-py2.py3-none-any.whl (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 500.9 kB/s eta 0:00:00
ERROR: Could not find a version that satisfies the requirement pyarmor.cli.core~=3.2.0 (from pyarmor) (from versions: none)
ERROR: No matching distribution found for pyarmor.cli.core~=3.2.0

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: pip install --upgrade pip
#
```

**Reply by jondy** [reply]
Sorry, the source isn't open, there are 2 solutions:
1. wait for Pyarmor 8 supports FreeBSD in future, it won't be supported soon, because there are some other upgrading work need to do.
2. using Pyarmor 7, it supports FreeBSD, but it's totally different from Pyarmor 8 and need purchase old license for Pyarmor 7, the new license doesn't work for Pyarmor 7

**Comment by Soujanya459** [comment]
I tried installing pyarmor 7.7.4 --> installation successful but error while registering the license

```
# pyarmor register pyarmor-regcode-1.txt
ERROR    Python 3.11+ is not supported now
# pyarmor -v
PyArmor Trial Version 7.7.4

PyArmor is a command line tool used to obfuscate python scripts, bind
obfuscated scripts to fixed machine or expire obfuscated scripts.

For more information, refer to https://pyarmor.readthedocs.io
```

I have python 3.11.2 in FreeBSD machine .. 
can you please suggest what is the correct pyarmor version I should install in FreeBSD and I have pyarmor8 and parmor6 license.

**Reply by jondy** [reply]
Pyarmor 7 doesn't support Python 3.11

If you must use Python 3.11 in FreeBSD, there is no solution now.

Please check the latest Pyarmor documentation for supporting platforms and Python versions.

**Comment by Soujanya459** [comment]
Hi Jondy ,

as we have license for pyarmor8..
Is there ETA  for pyarmor8 support in FreeBSD ?

**Reply by jondy** [reply]
No exact date, around in 2 months.

**Comment by jondy** [comment]
@joonis @Soujanya459 would you please show `python3-config --ldflags`  in freebsd? I'd like to know lib python is shared or static link.

**Reply by Soujanya459** [reply]
Please find below response.. 
`
 python3-config --ldflags`
 `-L/var/crash/project/usr/local/lib  -ldl  -lutil -lm
`

**Comment by jondy** [comment]
Got it. Now FreeBSD has been supported by Pyarmor 8

**Comment by Soujanya459** [comment]
Thanks Jondy..
I'm able to install pyarmor 8.2.6 version and register the license.. but while doing obfuscation getting below error..

(ese_virtualenv) onefs-soujanya-1# pyarmor gen -r -O <<outDir>> dreamcatcher/lib/pyDreamCatcher/__main__.py
INFO     Python 3.11.2
INFO     Pyarmor 8.2.6 (pro), 005098, non-profits(TBD)
INFO     Platform freebsd.x86_64
INFO     search inputs ...
INFO     find script dreamcatcher/lib/pyDreamCatcher/__main__.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'freebsd.amd64'}
ERROR    please check all supported platforms in documentation "References"
ERROR    no found prebuilt runtime extension for platform "freebsd.amd64"
(ese_virtualenv) onefs-soujanya-1#

(ese_virtualenv) onefs-soujanya-1# uname -a
FreeBSD onefs-soujanya-1 12.0-CURRENT FreeBSD 12.0-CURRENT amd64
(ese_virtualenv) onefs-soujanya-1#

**Reply by jondy** [reply]
Got it, it will be fixed in next release. A quick workaround, try to use extra option `--platform` to obfuscate scripts
```
pyarmor gen --platform freebsd.x86_64 foo.py
```

**Reply by Soujanya459** [reply]
Thank You jondy .. it worked for me.. 

When this fix will be available .. ?can you please let me know the release version..

**Comment by jondy** [comment]
The latest version 8.2.8 should be OK
